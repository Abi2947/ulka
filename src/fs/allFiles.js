const fs = require("fs");
const path = require("path");

const allFiles = function (dirPath, ext, arrayOfFiles) {
  const files = fs.readdirSync(dirPath);

  arrayOfFiles = arrayOfFiles || [];

  files.forEach((file) => {
    const pathTo = path.join(dirPath, file);
    if (fs.statSync(pathTo).isDirectory()) {
      arrayOfFiles = allFiles(pathTo, ext, arrayOfFiles);
    } else {
      if (!ext || file.endsWith(ext)) arrayOfFiles.push(pathTo);
    }
  });

  return arrayOfFiles;
};

module.exports = allFiles;
