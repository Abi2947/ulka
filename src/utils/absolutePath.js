const path = require('path')

const absolutePath = args => {
  if (typeof args !== 'string') {
    throw new Error('Path provided should be string')
  }
  return path.join(process.cwd(), ...args.split('/'))
}

module.exports = absolutePath
