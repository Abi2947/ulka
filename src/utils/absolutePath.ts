import path from 'path'

const absolutePath = (args: string) => {
  if (typeof args !== 'string') {
    throw new Error('Path provided should be string')
  }
  return path.join(process.cwd(), ...args.split('/'))
}

export default absolutePath
