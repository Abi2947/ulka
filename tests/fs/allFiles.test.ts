import path from "path"
import allFiles from "../../src/fs/allFiles"
import absolutePath from "../../src/utils/absolutePath"

describe("all Files function", () => {
  test("should return all files in a folder", () => {
    const allFilesList = allFiles(
      absolutePath("tests/resources/fs_allfiles_test")
    ).map(f => path.relative(process.cwd(), f).split(path.sep).join("/"))
    expect(allFilesList).toMatchSnapshot()
  })

  test("should return ulka files only", () => {
    const allUlkaFiles = allFiles(
      absolutePath("tests/resources/fs_allfiles_test"),
      ".ulka"
    ).map(f => path.relative(process.cwd(), f).split(path.sep).join("/"))
    expect(allUlkaFiles).toMatchSnapshot()
  })

  test("should return js files only", () => {
    const allJsFiles = allFiles(
      absolutePath("tests/resources/fs_allfiles_test"),
      ".js"
    ).map(f => path.relative(process.cwd(), f).split(path.sep).join("/"))
    expect(allJsFiles).toMatchSnapshot()
  })
})
