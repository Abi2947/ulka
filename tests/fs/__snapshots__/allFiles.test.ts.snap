// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all Files function should return all files in a folder 1`] = `
Array [
  "tests/resources/fs_allfiles_test/file1.js",
  "tests/resources/fs_allfiles_test/file1.ulka",
  "tests/resources/fs_allfiles_test/file2.js",
  "tests/resources/fs_allfiles_test/file2.ulka",
  "tests/resources/fs_allfiles_test/folder1/folder1-file1.ulka",
  "tests/resources/fs_allfiles_test/folder1/folder2-file2.ulka",
]
`;

exports[`all Files function should return js files only 1`] = `
Array [
  "tests/resources/fs_allfiles_test/file1.js",
  "tests/resources/fs_allfiles_test/file2.js",
]
`;

exports[`all Files function should return ulka files only 1`] = `
Array [
  "tests/resources/fs_allfiles_test/file1.ulka",
  "tests/resources/fs_allfiles_test/file2.ulka",
  "tests/resources/fs_allfiles_test/folder1/folder1-file1.ulka",
  "tests/resources/fs_allfiles_test/folder1/folder2-file2.ulka",
]
`;
